### DISCUSSION TITLE
Highlight other usages of word under cursor

### DISCUSSION THREAD

**Discussion by matoous** [helix-editor/helix#1327] [Ideas] [answered]
I am wondering how would functionality of _highlighting other usage of word under cursor_ work with helix and whether it is even possible? For example see: https://github.com/RRethy/vim-illuminate. I guess other usages shouldn't be _selected_ but just _highlighted_ which might be confusing, maybe two types of highlight would be necessary, one lighter for this use-case (not big fan of that)? Open question.

**Comment by sudormrfbin** [comment, marked as answer]
We already have different highlights for primary and secondary selections, and adding one more highlight might make it confusing. I'm not sure if we want this in core and whether this is better suited as a plugin, but a way of potentially solving it is to draw a "box" around the same words (IIRC VSCode does it this way).
